stock_picking_putaway_recompute/models/stock_picking.py

```diff
@@ -23,15 +23,17 @@ def action_recompute_putaways(self) -> dict:
         for picking in self:
             picking.move_line_ids._recompute_putaways()
 
+    def _can_recompute_putaway(self):
+        self.ensure_one()
+        return self.picking_type_id.allow_to_recompute_putaways and self.state == "assigned" and not self.printed
+
     def _filtered_can_recompute_putaways(self) -> Self:
         """
         Filter current recordset in order to get the pickings that can
         """
-        return self.filtered(
-            lambda picking: picking.state == "assigned" and not picking.printed
-        )
+        return self.filtered(lambda pick: pick._can_recompute_putaway())
 
-    @api.depends("state", "printed")
+    @api.depends("picking_type_id.allow_to_recompute_putaways", "state", "printed")
```

Comment:
Don't recompute all pickings when config is changed
```suggestion
    @api.depends("state", "printed")
```

Reply:
@jbaudoux in fact, the filed is not stored. So, the computation will occur on first access. This change does not harm IMHO.

Reply:
ok then :)